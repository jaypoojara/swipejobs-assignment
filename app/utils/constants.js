export const NOTIFICATION_TYPE = {
  DEFAULT: 0,
  SUCCESS: 1,
  ERROR: 2,
  WARNING: 3,
  INFO: 4
};

export const WORKER_ID = '7f90df6e-b832-44e2-b624-3143d428001f';
