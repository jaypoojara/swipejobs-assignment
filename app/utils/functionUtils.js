import moment from 'moment';
import momentTz from 'moment-timezone';

export const getDuration = (startDate, endDate) => {
  const days = moment(startDate)
    .local()
    .format('MMM DD, ddd HH:mm A');
  const endDateTemp = moment(endDate)
    .local()
    .format('HH:mm A');
  const timeZone = momentTz.tz.guess();
  const timeZoneAbr = momentTz.tz(timeZone).zoneAbbr();
  return `${days} - ${endDateTemp} ${timeZoneAbr}`;
};
