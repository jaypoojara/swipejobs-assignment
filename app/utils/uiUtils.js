import { showMessage } from 'react-native-flash-message';
import { NOTIFICATION_TYPE } from './constants';
export function showMessageToUser({ title, message, notificationType }) {
  let messageType;
  switch (notificationType) {
    case NOTIFICATION_TYPE.DEFAULT:
      messageType = 'default';
      break;
    case NOTIFICATION_TYPE.INFO:
      messageType = 'info';
      break;
    case NOTIFICATION_TYPE.SUCCESS:
      messageType = 'success';
      break;
    case NOTIFICATION_TYPE.ERROR:
      messageType = 'danger';
      break;
    case NOTIFICATION_TYPE.WARNING:
      messageType = 'warning';
      break;
    default:
      messageType = 'none';
  }
  setTimeout(() => {
    showMessage({
      message: title,
      description: message,
      type: messageType
    });
  }, 100);
}
