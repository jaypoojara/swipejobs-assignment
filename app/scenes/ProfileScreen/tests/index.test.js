import React from 'react';
import { renderProvider } from 'app/utils/testUtils';
import { ProfileScreenTest } from '../index';

describe('<ProfileScreen /> Container tests', () => {
  const users = {
    address: {
      formattedAddress: '1 Downing St, Chicago, IL 60654, USA',
      zoneId: 'America/Chicago'
    },
    email: 'jim.rose@gmail.com',
    firstName: 'Jim',
    lastName: 'Rose',
    maxJobDistance: 20.02,
    phoneNumber: '5096290220',
    workerId: '7f90df6e-b832-44e2-b624-3143d428001f'
  };
  it('should render and match the ProfileScreen snapshot', () => {
    const baseElement = renderProvider(<ProfileScreenTest user={users} />);
    expect(baseElement).toMatchSnapshot();
  });
});
