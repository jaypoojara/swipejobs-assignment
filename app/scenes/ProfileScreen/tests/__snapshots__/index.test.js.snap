// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileScreen /> Container tests should render and match the ProfileScreen snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "marginHorizontal": 10,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "flex": 1,
                "fontSize": 20,
                "marginRight": 10,
                "textAlign": "center",
              },
            ]
          }
          testID="t"
        >
          Jim Rose
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Object {
        "margin": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
      testID="t"
    >
      First Name
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
        }
      }
      testID="t"
    >
      Jim
    </Text>
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
        testID="t"
      >
        Last Name
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
          }
        }
        testID="t"
      >
        Rose
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
        testID="t"
      >
        Address
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
          }
        }
        testID="t"
      >
        1 Downing St, Chicago, IL 60654, USA
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
        testID="t"
      >
        Job Distance
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
          }
        }
        testID="t"
      >
        20.02 miles
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
        testID="t"
      >
        Phone Number
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
          }
        }
        testID="t"
      >
        5096290220
      </Text>
    </View>
  </View>
</View>
`;
